File: 53-maximum-subarray/53-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSub = nums[0];
    let currentSum = 0;
    for(i=0; i<nums.length; i++) {
        if(currentSum < 0) currentSum = 0;
        currentSum += nums[i];
        maxSub = Math.max(maxSub, currentSum)
    }
    return maxSub;
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total = 0
        max_total = nums[0]
        
        for value in nums:
            if total < 0:
                total = 0
            total += value
            max_total = max(total, max_total)
                
        return max_total